File: programmers/Python/Level2/스킬트리.py
def correct_sequence(skill, tree):
    result = ""
    for c in tree:
        if c in skill:
            result += c
    return result == skill[:len(result)]


def solution(skill, skill_trees):
    answer = 0
    for skill_tree in skill_trees:
        if correct_sequence(skill, skill_tree):
            answer += 1
    return answer

File: programmers-coding-test/C++/Level2/스킬트리.cpp
#include <string>
#include <vector>

using namespace std;

bool isCorrect(string skill, string tree) {

    // skill에 포함되는 스킬이 tree에 없거나, 
    // new_skill이 skill의 시작 sub string인 경우에 counting

    string new_skill = "";
    for (int i = 0; i < tree.length(); i++)
    {
        if (skill.find(tree[i]) != -1) 
            new_skill.append(tree.substr(i, 1));
    }

    return new_skill == "" || skill.find(new_skill) == 0;
}

int solution(string skill, vector<string> skill_trees) {

    int answer = 0;
    for (string tree : skill_trees)
        if (isCorrect(skill, tree)) answer++;

    return answer;
}
